tests/stub/routing/test_routing_v5x0.py

```diff
@@ -1873,6 +1873,59 @@ def work(tx):
         self.assertEqual([[]], sequences)
         self.assertEqual(2, try_count)
 
+    def test_should_forget_router_address_on_database_unavailable_error(self):
+        # TODO remove this block once all languages work
+        if get_driver_name() in ["go"]:
+            self.skipTest("requires investigation")
+        driver = Driver(self._backend, self._uri_with_context, self._auth,
+                        self._userAgent)
+        self.start_server(self._routingServer1,
+                          "router_yielding_writer1.script")
+        self.start_server(
+            self._writeServer1,
+            "writer_tx_yielding_database_unavailable_failure.script"
+        )
+        self.start_server(
+            self._routingServer2,
+            "router_yielding_database_unavailable_failure.script"
+        )
+        self.start_server(self._writeServer2, "writer_tx.script")
+
+        session = driver.session("w", database=self.adb)
+        sequences = []
+        try_count = 0
+
+        def work(tx):
+            nonlocal try_count
+            try_count = try_count + 1
+            result = tx.run("RETURN 1 as n")
+            sequences.append(self.collect_records(result))
+
+        retried = False
+
+        def on_retryable_negative_hook(_):
+            nonlocal retried
+            if not retried:
+                self._routingServer1.done()
```

Comment:
You could probably also assert that `self._writeServer1.done()` here already.

Reply:
I don't know if the driver already release the connection back to the pool at this point in time